{
  "Software_Repo_MappingId": null,
  "Software_RepoId": "f6b863cd-8f1c-4583-8bc5-ee59e86e5cf1",
  "Software_Name": "Python 3.12.5 (64-bit)",
  "CVEID": "CVE-2023-40217",
  "Summary": "An issue was discovered in Python before 3.8.18, 3.9.x before 3.9.18, 3.10.x before 3.10.13, and 3.11.x before 3.11.5. It primarily affects servers (such as HTTP servers) that use TLS client authentication. If a TLS server-side socket is created, receives data into the socket buffer, and then is closed quickly, there is a brief window where the SSLSocket instance will detect the socket as \"not connected\" and won't initiate a handshake, but buffered data will still be readable from the socket buffer. This data will not be authenticated if the server-side TLS peer is expecting client certificate authentication, and is indistinguishable from valid TLS stream data. Data is limited in size to the amount that will fit in the buffer. (The TLS connection cannot directly be used for data exfiltration because the vulnerable code path requires that the connection be closed on initialization of the SSLSocket.)",
  "LastUpdate": 0,
  "CVEE": {
    "CVEEId": "CVE-2023-40217",
    "CVSS_V2_baseScore": null,
    "CVSS_V2_Data": null,
    "CVSS_V3_baseScore": 5.3,
    "CVSS_V3_Data": "{\"version\":\"3.1\",\"baseScore\":5.3,\"accessVector\":null,\"vectorString\":\"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:N/A:N\",\"authentication\":null,\"integrityImpact\":\"NONE\",\"accessComplexity\":null,\"availabilityImpact\":\"NONE\",\"confidentialityImpact\":\"LOW\",\"scope\":\"UNCHANGED\",\"attackVector\":\"NETWORK\",\"baseSeverity\":\"MEDIUM\",\"userInteraction\":\"NONE\",\"attackComplexity\":\"LOW\",\"privilegesRequired\":\"NONE\"}",
    "PublishedDate": "2023-08-25T01:15:09.017",
    "ModifyDate": "2024-11-21T08:19:01.303",
    "VulnStatus": "Modified",
    "JsonData": "{\"id\": \"CVE-2023-40217\", \"summary\": \"An issue was discovered in Python before 3.8.18, 3.9.x before 3.9.18, 3.10.x before 3.10.13, and 3.11.x before 3.11.5. It primarily affects servers (such as HTTP servers) that use TLS client authentication. If a TLS server-side socket is created, receives data into the socket buffer, and then is closed quickly, there is a brief window where the SSLSocket instance will detect the socket as \\\"not connected\\\" and won't initiate a handshake, but buffered data will still be readable from the socket buffer. This data will not be authenticated if the server-side TLS peer is expecting client certificate authentication, and is indistinguishable from valid TLS stream data. Data is limited in size to the amount that will fit in the buffer. (The TLS connection cannot directly be used for data exfiltration because the vulnerable code path requires that the connection be closed on initialization of the SSLSocket.)\", \"created_at\": \"2023-08-25T01:15:09Z\", \"updated_at\": \"2024-11-21T08:19:01Z\", \"cvss\": {\"v2\": null, \"v3\": 5.3}, \"vendors\": {\"python\": [\"python\"]}, \"cwes\": [\"NVD-CWE-noinfo\"], \"raw_nvd_data\": {\"id\": \"CVE-2023-40217\", \"metrics\": {\"cvssMetricV31\": [{\"type\": \"Primary\", \"source\": \"nvd@nist.gov\", \"cvssData\": {\"scope\": \"UNCHANGED\", \"version\": \"3.1\", \"baseScore\": 5.3, \"attackVector\": \"NETWORK\", \"baseSeverity\": \"MEDIUM\", \"vectorString\": \"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:N/A:N\", \"integrityImpact\": \"NONE\", \"userInteraction\": \"NONE\", \"attackComplexity\": \"LOW\", \"availabilityImpact\": \"NONE\", \"privilegesRequired\": \"NONE\", \"confidentialityImpact\": \"LOW\"}, \"impactScore\": 1.4, \"exploitabilityScore\": 3.9}]}, \"published\": \"2023-08-25T01:15:09.017\", \"references\": [{\"url\": \"https://lists.debian.org/debian-lts-announce/2023/09/msg00022.html\", \"source\": \"cve@mitre.org\"}, {\"url\": \"https://lists.debian.org/debian-lts-announce/2023/10/msg00017.html\", \"source\": \"cve@mitre.org\"}, {\"url\": \"https://mail.python.org/archives/list/security-announce%40python.org/thread/PEPLII27KYHLF4AK3ZQGKYNCRERG4YXY/\", \"source\": \"cve@mitre.org\"}, {\"url\": \"https://security.netapp.com/advisory/ntap-20231006-0014/\", \"source\": \"cve@mitre.org\"}, {\"url\": \"https://www.python.org/dev/security/\", \"tags\": [\"Vendor Advisory\"], \"source\": \"cve@mitre.org\"}, {\"url\": \"https://lists.debian.org/debian-lts-announce/2023/09/msg00022.html\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://lists.debian.org/debian-lts-announce/2023/10/msg00017.html\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://mail.python.org/archives/list/security-announce%40python.org/thread/PEPLII27KYHLF4AK3ZQGKYNCRERG4YXY/\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://security.netapp.com/advisory/ntap-20231006-0014/\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://www.python.org/dev/security/\", \"tags\": [\"Vendor Advisory\"], \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}], \"vulnStatus\": \"Modified\", \"weaknesses\": [{\"type\": \"Primary\", \"source\": \"nvd@nist.gov\", \"description\": [{\"lang\": \"en\", \"value\": \"NVD-CWE-noinfo\"}]}], \"descriptions\": [{\"lang\": \"en\", \"value\": \"An issue was discovered in Python before 3.8.18, 3.9.x before 3.9.18, 3.10.x before 3.10.13, and 3.11.x before 3.11.5. It primarily affects servers (such as HTTP servers) that use TLS client authentication. If a TLS server-side socket is created, receives data into the socket buffer, and then is closed quickly, there is a brief window where the SSLSocket instance will detect the socket as \\\"not connected\\\" and won't initiate a handshake, but buffered data will still be readable from the socket buffer. This data will not be authenticated if the server-side TLS peer is expecting client certificate authentication, and is indistinguishable from valid TLS stream data. Data is limited in size to the amount that will fit in the buffer. (The TLS connection cannot directly be used for data exfiltration because the vulnerable code path requires that the connection be closed on initialization of the SSLSocket.)\"}], \"lastModified\": \"2024-11-21T08:19:01.303\", \"configurations\": [{\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:a:python:python:*:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"57315FC0-A50B-4677-9186-FD04F046419C\", \"versionEndExcluding\": \"3.8.18\"}, {\"criteria\": \"cpe:2.3:a:python:python:*:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"9AEA72AE-65B5-4685-BB0C-5E077D4BFA6C\", \"versionEndExcluding\": \"3.9.18\", \"versionStartIncluding\": \"3.9.0\"}, {\"criteria\": \"cpe:2.3:a:python:python:*:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"23CCE2C2-CC19-4C91-A173-B583E598FAFD\", \"versionEndExcluding\": \"3.10.13\", \"versionStartIncluding\": \"3.10.0\"}, {\"criteria\": \"cpe:2.3:a:python:python:*:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"2EE37B61-ADAD-4F98-B732-DEBC5BECBC55\", \"versionEndExcluding\": \"3.11.5\", \"versionStartIncluding\": \"3.11.0\"}], \"operator\": \"OR\"}]}], \"sourceIdentifier\": \"cve@mitre.org\"}}\n",
    "LastUpdate": 0,
    "VendorL": [
      "python"
    ],
    "ProductL": [
      "python"
    ],
    "Software_Repo_Mappings": []
  },
  "Software_Repo_Mapping_Conditions": [
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
      "VersionEndExcluding": "3.8.18",
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": null,
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    },
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
      "VersionEndExcluding": "3.9.18",
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": "3.9.0",
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    },
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
      "VersionEndExcluding": "3.10.13",
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": "3.10.0",
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    },
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
      "VersionEndExcluding": "3.11.5",
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": "3.11.0",
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    }
  ]
}